sdk/appcenter/src/main/java/com/microsoft/appcenter/persistence/DatabasePersistence.java

```diff
@@ -286,16 +299,27 @@ public long putLog(@NonNull Log log, @NonNull String group, @IntRange(from = Fla
                 targetToken = null;
             }
             long maxSize = mDatabaseManager.getMaxSize();
-            if (maxSize == -1) {
+            if (maxSize == OPERATION_FAILED_FLAG) {
                 throw new PersistenceException("Failed to store a log to the Persistence database.");
             }
             if (!isLargePayload && maxSize <= payloadSize) {
```

Comment:
This condition allows writing a too large file. We should remove `!isLargePayload` condition.